crates/findex/src/implementations/redis/findex.rs

```diff
@@ -0,0 +1,434 @@
+        trace!("fetch_entry_table non empty tuples len: {}", tuples.len(),);
+
+        Ok(EncryptedMultiTable(tuples))
+    }
+
+    async fn fetch_chain_table(
+        &self,
+        chain_table_uids: Uids<UID_LENGTH>,
+    ) -> Result<EncryptedTable<UID_LENGTH>, FindexRedisError> {
+        trace!(
+            "fetch_chain_table num entries: {}:",
+            chain_table_uids.0.len(),
+        );
+
+        //guard against empty uids
+        if chain_table_uids.0.is_empty() {
+            return Ok(EncryptedTable::default());
+        }
+
+        let keys: Vec<Vec<u8>> = chain_table_uids
+            .0
+            .iter()
+            .map(|uid| key(FindexTable::Chain, uid))
+            .collect();
+        let values: Vec<Vec<u8>> = self.mgr.clone().mget(keys).await?;
+
+        Ok(EncryptedTable::from(
+            chain_table_uids
+                .0
+                .into_iter()
+                .zip(values)
+                .filter(|(_uid, v)| !v.is_empty())
+                .collect::<HashMap<Uid<UID_LENGTH>, Vec<u8>>>(),
+        ))
+    }
+
+    async fn upsert_entry_table(
+        &self,
+        modifications: UpsertData<UID_LENGTH>,
+    ) -> Result<EncryptedTable<UID_LENGTH>, FindexRedisError> {
+        // Prevent upserting while compacting
+        if self.compact_lock.try_read().is_err() {
+            return Err(FindexRedisError::Compacting);
+        }
+
+        trace!("upsert_entry_table num keywords {:?}", modifications.len());
+
+        let mut rejected = EncryptedTable::default();
+        for (uid, (old_value, new_value)) in modifications {
+            let value: Vec<u8> = self
+                .upsert_script
+                .arg(key(FindexTable::Entry, &uid))
+                .arg(old_value.unwrap_or_default())
+                .arg(new_value)
+                .invoke_async(&mut self.mgr.clone())
+                .await?;
+            if !value.is_empty() {
+                rejected.insert(uid, value);
+            }
+        }
+        trace!("upsert_entry_table rejected: {}", rejected.len());
+        Ok(rejected)
+    }
+
+    async fn insert_chain_table(
+        &self,
+        items: EncryptedTable<UID_LENGTH>,
+    ) -> Result<(), FindexRedisError> {
+        let mut pipe = pipe();
+        for item in items {
+            pipe.set(key(FindexTable::Chain, &item.0), item.1);
+        }
+        pipe.atomic().query_async(&mut self.mgr.clone()).await?;
+        Ok(())
+    }
+
+    async fn update_lines(
+        &self,
+        chain_table_uids_to_remove: Uids<UID_LENGTH>,
+        new_encrypted_entry_table_items: EncryptedTable<UID_LENGTH>,
+        new_encrypted_chain_table_items: EncryptedTable<UID_LENGTH>,
+    ) -> Result<(), FindexRedisError> {
+        // Prevent compacting while already compacting
+        // We could add support to make sure we are not upserting
+        if self.compact_lock.try_read().is_err() {
+            return Err(FindexRedisError::Compacting);
+        }
```

Comment:
Same here, I think the problem is only partially solved. Another action needs to be taken anyway so I would remove this.

Reply:
see comment above